### ISSUE TITLE
Adam Optimizer

### ISSUE THREAD

**Issue by julie1** [Kamnitsask/deepmedic#125]
I needed to change three lines of code in the deepmedic/deepmedic/neuralnet/optimizers.py
file to get the Adam optimizer to work.  I replaced : 
def get_update_ops_given_grads(self, cost, params) :
        # Epsilon on paper was 10**(-8).
        # Code is on par with version V8 of Kingma's paper.
        grads = tf.gradients(cost, params)
        
        updates = []
        
        i = self._i_adam
        i_t = i + 1.
        fix1 = 1. - (self._b1_adam)**i_t
        fix2 = 1. - (self._b2_adam)**i_t
        lr_t = self._learning_rate * (tf.sqrt(fix2) / fix1)
        for param, grad, m, v in zip(params, grads, self._means_of_grads, self._vars_of_grads):

in the class AdamOptimizer(Optimizer)
by : 
def get_update_ops_given_grads(self, grads) :
        # Epsilon on paper was 10**(-8).
        # Code is on par with version V8 of Kingma's paper.
      #  grads = tf.gradients(cost, params)
        
        updates = []
        
        i = self._i_adam
        i_t = i + 1.
        fix1 = 1. - (self._b1_adam)**i_t
        fix2 = 1. - (self._b2_adam)**i_t
        lr_t = self._learning_rate * (tf.sqrt(fix2) / fix1)
        for param, grad, m, v in zip(self._params_to_opt, grads, self._means_of_grads, self._vars_of_grads):

Thank you.

**Comment by Kamnitsask** [comment]
Thank you Julie,

I havent used Adam in a long while. I got better results from the other ones (sgd/rmsprop) so I stick with them. So I missed this.

I will go back to it and update the code in the next batch of fixes. Thanks.

Lets leave this issue on for now till I apply it.

Thanks!

**Comment by Kamnitsask** [comment]
Resolved in commit 88e068e, thanks a lot for bringing this to my attention.
